master/internal/kubernetes/pod_test.go

```diff
@@ -0,0 +1,795 @@
+	podMap["task"].Purge()
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+
+	// Pod status Pending, Pod Scheduled
+	t.Logf("Testing pod scheduled with pending status")
+	typeMeta := metaV1.TypeMeta{Kind: "rest test"}
+	objectMeta := metaV1.ObjectMeta{
+		Name: "test meta",
+	}
+	condition := k8sV1.PodCondition{
+		Type:    k8sV1.PodScheduled,
+		Status:  k8sV1.ConditionTrue,
+		Message: "This doesn't matter :)",
+	}
+	status := k8sV1.PodStatus{
+		Phase:      k8sV1.PodPending,
+		Conditions: []k8sV1.PodCondition{condition},
+	}
+	pod := k8sV1.Pod{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Status:     status,
+	}
+	statusUpdate := podStatusUpdate{updatedPod: &pod}
+
+	system.Ask(ref, statusUpdate)
+	time.Sleep(time.Second)
+
+	assert.Equal(t, podMap["task"].GetLength(), 2)
+	assert.Equal(t, newPod.container.State, container.Starting)
+	podMap["task"].Purge()
+	system.Ask(ref, statusUpdate)
+	time.Sleep(time.Second)
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+	assert.Equal(t, newPod.container.State, container.Starting)
+
+	// Pod status Running, but container status Waiting
+	t.Logf("Testing pod running with waiting status")
+	system, newPod, ref, podMap, _ = createPodWithMockQueue()
+	podMap["task"].Purge()
+	assert.Equal(t, podMap["task"].GetLength(), 0)
+
+	containerStatuses := []k8sV1.ContainerStatus{
+		{
+			Name:  "determined-container",
+			State: k8sV1.ContainerState{Waiting: &k8sV1.ContainerStateWaiting{}},
+		},
+	}
+	status = k8sV1.PodStatus{
+		Phase:             k8sV1.PodRunning,
+		ContainerStatuses: containerStatuses,
+	}
+	pod = k8sV1.Pod{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Status:     status,
+	}
+	statusUpdate = podStatusUpdate{updatedPod: &pod}
+
+	system.Ask(ref, statusUpdate)
+	time.Sleep(time.Second)
+	assert.Equal(t, podMap["task"].GetLength(), 2)
+	assert.Equal(t, newPod.container.State, container.Starting)
+
+	// Pod status running, but no Container State inside
+	t.Logf("Testing pod running with no status")
+	system, newPod, ref, podMap, _ = createPodWithMockQueue()
+	podMap["task"].Purge()
+	status = k8sV1.PodStatus{
+		Phase:             k8sV1.PodRunning,
+		ContainerStatuses: []k8sV1.ContainerStatus{{Name: "determined-container"}},
+	}
+	pod = k8sV1.Pod{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Status:     status,
+	}
+	statusUpdate = podStatusUpdate{updatedPod: &pod}
+	system.Ask(ref, statusUpdate)
+	time.Sleep(time.Second)
+	assert.Equal(t, podMap["task"].GetLength(), 2)
+	assert.Equal(t, newPod.container.State, container.Starting)
+
+	err := os.Remove("k8_init_container_entrypoint.sh")
+	if err != nil {
+		t.Logf("Failed to remove entrypoint")
+	}
+}
+
+func TestMultipleContainersRunning(t *testing.T) {
+	// Status update test involving two containers
+	setupEntrypoint(t)
+
+	// If one of the containers doesn't have running state, nothing changes
+	t.Logf("Testing two pods and one doesn't have running state")
```

Comment:
should it be `Testing  pod with two containers and one doesn't have running state`?